readthedocs/doc_builder/environments.py

```diff
@@ -103,11 +103,9 @@ def __init__(
         if cwd is None:
             cwd = os.getcwd()
         self.cwd = cwd
-        self.environment = os.environ.copy()
```

Comment:
Do you know _why_ we were using all the env variables from the host here previously?

Reply:
Because we were only using a local environment, it wasn't considering using another one like docker.

Reply:
Yeah, I mean, why did we need _all_ the variables from the host?

Reply:
It was needed for the local environment only (and we were setting it for both, docker and local, it wasn't given any error bc we were using a local environment only for checkout, now we can use docker), now we are setting that to in the local environment only, not in docker.

Reply:
A user reported (https://github.com/readthedocs/readthedocs.org/issues/6577) a case where this is a problem. He uses `http_proxy` to clone the repositories. With this change, we are not passing any of these variables to the environment anymore.